### PR TITLE
feat: add Slurm/Spur GPU runner

### PR DESCRIPTION
**Pull request by irvineoy**
## Summary

- add a Slurm/Spur wrapper that allocates GPUs on a compute node, then reuses the existing Docker benchmark runner
- support a 1-GPU development path and an 8-GPU parallel path using Spur-provided GPU visibility variables
- support standalone Codex installations plus isolated Codex, Claude Code, and Cursor Agent credentials inside workers
- add Make targets, runner tests, and usage documentation for interactive and asynchronous jobs
- avoid `/dev/mem` passthrough, which is rejected by the cluster Docker authorization policy

## Validation

- `make check-docker-runner`
- `make check-slurm-runner`
- `bash -n src/scripts/slurm_benchmark.sh src/scripts/docker_benchmark.sh`
- `git diff --check`
- full non-GPU suite in the ROCm container on a compute node: `464 passed, 4 skipped` (Spur job `111733`)
- 1-GPU MI355X smoke test: Spur job `111267`
- Codex, Claude Code, and Cursor Agent login checks in an allocated Docker container: Spur job `111268`
- 8-GPU parallel smoke test: Spur job `111270`
- asynchronous `slurm-submit` orchestration: Spur job `111278`
- 8-worker `slurm-parallel-run` orchestration: Spur job `111281`
- real 12-task Codex end-to-end run on 8 MI355X GPUs: Spur job `123618`, `COMPLETED`, exit `0:0`, 40m40s
  - all 8 workers mapped to distinct physical GPU IDs `0..7`
  - dynamic queue drained from 12 tasks, with workers claiming follow-up tasks after their first completion
  - 12/12 centralized compilation and correctness checks passed; 12 result files and the aggregate CSV were generated
  - 11/12 tasks produced valid matched performance measurements; MaskedLanguageModel compiled and passed correctness but its optimized performance command exited 1, so it was recorded without a valid speedup

### CODE REVIEW

**Comment by irvineoy** [comment]
## Slurm 8-GPU end-to-end validation

I ran an end-to-end Slurm validation of this PR using real AgentKernelArena tasks and the Codex agent. This was not limited to a container smoke test or a `--help` invocation.

### Test setup

- Slurm job: `123618`
- Partition: `amd-spur`
- Account: `amd-aicos`
- QoS selected by Slurm: `amd-aicos-qos`
- Allocation: 1 node, 8 x AMD Instinct MI355X GPUs, 64 CPUs, 512 GB RAM
- Compute node: `crsuse2-m2m-243`
- Container image: `lmsysorg/sglang-rocm:v0.5.14-rocm720-mi35x-20260705`
- Agent: `codex` using the existing authenticated CLI state
- Target architecture: `MI355X`
- Run suffix: `pr95_slurm8_e2e_12tasks`

The test was submitted with the new Slurm parallel target:

```bash
make slurm-parallel-submit \
  CONFIG=config_slurm_e2e_12tasks_mi355x.yaml \
  SLURM_ACCOUNT=amd-aicos \
  SLURM_PARALLEL_TIME=04:00:00 \
  RUN_ARGS="--run-suffix pr95_slurm8_e2e_12tasks"
```

The temporary run config contained 12 existing HIP tasks:

1. CrossEntropyLossLabelSmoothing
2. Feedforward
3. FusedLeakyReLU
4. GateGRUSelectionLayer
5. GELU
6. InnerProd
7. ItemQueryAttention
8. KDLoss
9. layer_normalization
10. MaskedLanguageModel
11. MLP_model
12. MultiHeadAttention

### What this exercised

- Slurm batch submission from the login node
- Allocation of all 8 requested MI355X GPUs
- Container startup and ROCm/PyTorch/Triton GPU detection
- Codex authentication and invocation inside the container
- Eight parallel workers bound to distinct physical GPUs `0` through `7`
- Dynamic queue scheduling with more tasks than GPUs
- Per-task agent execution, compilation, correctness evaluation, performance evaluation, result generation, and final report aggregation

The first eight tasks were claimed concurrently by the eight workers. As workers completed, four of them claimed the remaining tasks, confirming that the shared queue continues scheduling work beyond the initial one-task-per-GPU wave.

### Timing and scheduler result

- Slurm state: `COMPLETED`
- Exit code: `0:0`
- Runtime: `00:40:40`
- Queue result: `done=12`, `failed=0`
- Generated `task_result.yaml` files: 12
- Aggregate `overall_summary.csv`: generated successfully

The first use on the compute node included approximately four minutes to pull the pinned container image.

### Evaluation result

- Compilation: **12/12 passed**
- Correctness: **12/12 passed**
- Tasks with a valid matched performance result: **11/12**
- Tasks with speedup greater than 1: **10/12**
- Average reported speedup: **16.79x**
- Median reported speedup: **3.66x**

One task, `MaskedLanguageModel`, compiled successfully and passed centralized correctness evaluation, but its optimized performance subprocess exited with status 1 and did not produce a valid optimized timing. It was therefore reported as `0.0x` with a performance-method mismatch. The same worker/GPU successfully completed another task, the baseline measurement succeeded, and the overall Slurm job and queue completed normally, so this appears to be a task/candidate performance-run issue rather than a Slurm allocation or parallel scheduling failure.

There were also non-fatal Codex warnings about a stale absolute rollout/session path copied from the login environment. They did not prevent any agent task from running or completing, but tightening the copied Codex state could be considered separately.

### Conclusion

The PR successfully ran a real 12-task AgentKernelArena workload across 8 allocated MI355X GPUs under Slurm. The allocation, GPU binding, container execution, authenticated Codex invocation, dynamic multi-worker queue, centralized evaluation, and report aggregation all worked end to end. No Slurm integration blocker was found in this validation.

**Comment by wzhuk-amd** [comment]
lgtm, also verified a validation run on my end (only after diagnosing a spur node with broken docker setup)
